doschema/jsvalidation.py

```diff
@@ -0,0 +1,326 @@
+            if field and field.field_type != "object":
+                err_msg = "Dependencies {0} error in schema {1}."
+                raise JSONSchemaDependenciesException(
+                    self.uri,
+                    err_msg.format(
+                        self.make_json_pointer(curr_field),
+                        self.uri
+                    )
+                )
+            self.add_field_to_dict(curr_field, 'object')
+            self.validate_dependencies(curr_schema, curr_field)
+
+    def add_field_to_dict(self, curr_field, field_type):
+        """Add record to the dictionary.
+
+        :param curr_field: Tuple with path to currently processed field.
+        :param field_type: Type of currently processed field.
+        """
+        self.fields_types_dict[curr_field] = FieldToAdd(
+                schema_index=self.uri,
+                field_tuple=curr_field,
+                field_type=field_type
+            )
+
+    def validate_array(self, field_value, curr_field):
+        """Process each JSON object in the array according to ignore_index.
+
+        :param field_value: Type of the field.
+        :param curr_field: Tuple with path to currently processed field.
+        """
+        field_path = curr_field
+        if isinstance(field_value, list):
+            for elem in field_value:
+                if self.ignore_index:
+                    new_field_path = field_path
+                else:
+                    new_field_path = field_path + (field_value.index(elem), )
+
+                self.validate_schema(elem, new_field_path)
+
+        elif isinstance(field_value, dict):
+            self.validate_schema(field_value, curr_field)
+
+    def validate_collection(self, curr_schema, curr_field):
+        """Check schema for allOf, anyOf and oneOf.
+
+        :param curr_schema: Schema or subschema that is currently processed.
+        :param curr_field: Tuple with path to currently processed field.
+        """
+        keys = self.COLLECTION_KEYS.intersection(curr_schema)
+        for key in keys:
+            subschema = curr_schema[key]
+
+            if isinstance(subschema, list):
+                for elem in subschema:
+                    self.validate_schema(elem, curr_field)
+
+                props = map(lambda x: x.get('properties'), subschema)
+                flattened_props = {}
+
+                for prop in props:
+                    if prop:
+                        for key, value in prop.items():
+                            flattened_props[key] = value
+
+                for dependency in map(
+                    lambda x: x.get('dependencies'),
+                    subschema
+                ):
+                    if dependency:
+                        self.validate_dependencies(
+                            {'properties': props},
+                            curr_field + ('dependencies', )
+                        )
+            elif isinstance(subschema, dict):
```

Comment:
Can a collection value actually be a dict? It does not seem to work when I try it in a jsonschema linter.